app/src/main/admin/components/ModellingGroups/SingleGroup/Members/ModellingGroupMembersAdd.tsx

```diff
@@ -89,5 +75,24 @@ export class AddMember extends React.Component<Props, AddMemberState> {
             </div>
         </form>;
     }
+}
+
+export const mapStateToProps = (state: AdminAppState, props: Partial<ModellingGroupMembersAddProps>) :Partial<ModellingGroupMembersAddProps> => {
+    return {
+        members: props.members,
+        users: state.users.users,
+        groupId: props.groupId
+    }
+};
+
+export const mapDispatchToProps = (dispatch: Dispatch<AdminAppState>): Partial<ModellingGroupMembersAddProps> => {
+    return {
+        addUserToGroup: (groupId: string, username: string) => dispatch(modellingGroupsActionCreators.addUserToGroup(groupId, username))
```

Comment:
This is a big improvement :) looks like this component wasn't even under integration test before which was entirely my fault - this was one of the first react component I wrote!